File: 2026/06-June/16-Tuesday/solution.cs
public class Solution {
    public string ProcessStr(string s) {
        List<char> res = new List<char>();

        foreach (char c in s) {
            if (char.IsLower(c)) {
                res.Add(c);
            }
            else if (c == '*') {
                if (res.Count > 0)
                    res.RemoveAt(res.Count - 1);
            }
            else if (c == '#') {
                res.AddRange(res.ToArray());
            }
            else { // '%'
                res.Reverse();
            }
        }

        return new string(res.ToArray());
    }
}

File: 2026/06-June/16-Tuesday/solution.js
var processStr = function(s) {
    let res = [];

    for (const ch of s) {
        if (ch >= 'a' && ch <= 'z') {
            res.push(ch);
        } else if (ch === '*') {
            if (res.length) res.pop();
        } else if (ch === '#') {
            res.push(...res);
        } else { // '%'
            res.reverse();
        }
    }

    return res.join('');
};
